openmdao/core/group.py

```diff
@@ -941,7 +951,9 @@ def solve_linear(self, dumat, drmat, vois, mode=None, solver=None):
         if len(rhs_buf) == 0:
             return
 
+        solver.rel_inputs = rel_inputs
         sol_buf = solver.solve(rhs_buf, self, mode=mode)
+        solver.rel_inputs = None
```

Comment:
I could have made it a keyword argument, but then I would have had to cache it locally anyway in order for it to be passed through numpy fortran/c into the gmres callback function. I just cut out the middleman.